lib/blockchain_double.js

```diff
@@ -418,29 +418,35 @@ BlockchainDouble.prototype.sortByPriceAndNonce = function() {
   self.pending_transactions = sorted_transactions;
 };
 
-BlockchainDouble.prototype.processCall = function(tx, callback) {
+BlockchainDouble.prototype.processCall = function(tx, blockNumber, callback) {
   var self = this;
+  var startingStateRoot = self.stateTrie.root;
 
+  var cleanUpAndReturn = function (err, result, changeRoot) {
+    // For defaultBlock, undo state root changes
+    if (changeRoot){
+      self.stateTrie.root = startingStateRoot;
+    }
 
-  self.latestBlock(function (err, latestBlock) {
-    if (err) {
-      return callback(err);
+    self.vm.stateManager.revert(function (e) {
+      callback(err || e, result);
+    });
+  };
+
+  var runCall = function(tx, changeRoot, err, block){
+    // For defaultBlock, use that block's root
```

Comment:
Need to handle this err.

Reply:
Generally I feel like if we're defining functions in functions like this, we've got a cohesion problem (trying to do too much in one place).

Also we should probably be using `pify` or `util.promisify` to promisify our callback hell, as well.

Not sure whether to suggest this be changed, though - I'm like inches away from a big refactoring here, anyway. Might be easier to just solve it then.